### ISSUE TITLE
UiButton ripple not centered on cursor when clicking over button text/icon

### ISSUE THREAD

**Issue by joshuathompson** [JosephusPaye/Keen-UI#220]
On Ubuntu 16.04 and Chrome 56.

If you go to the [example page](https://josephuspaye.github.io/Keen-UI/#/ui-button) for the UiButton and click over an icon or text on a button then a majority of the time the ink ripple will start its animation off-center instead of directly on top of where was clicked.

**Comment by alinnert** [comment]
I've also noticed this in Chrome 56 and EdgeHTML 14 on Windows 10. It works in Firefox 51 though.

It looks like it's using the cursor position relative to the inside `div` element instead of the `button` element.

A shot in the dark: are you using `event.target` instead of `event.currentTarget` to get the coordinates?

**Comment by JosephusPaye** [comment]
Using `event.currentTarget`, but falls back to `event.target` (to support touch events) as seen here: https://github.com/JosephusPaye/Keen-UI/blob/master/src/UiRippleInk.vue#L13.

Have no idea why it's off though. Needs further debugging.

**Comment by alinnert** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/offsetX

The spec says `offsetX`/`offsetY` refer to `event.target` (at least it says "target node").

The fallback calculation in the linked file (the one using `getBoundingClientRect()` and `event.clientX`) should be the correct one. So it might be enough to remove the usage of `offsetX` etc.

**Comment by lbjones** [comment]
Perhaps a simple fix would be to not have the ripple start with a 0 radius.  If you see google's implementation of the ripple, you wouldn't notice if the ripple didn't start exactly where the mouse was centered.

http://material-components-web.appspot.com/button.html

**Comment by JosephusPaye** [comment]
Fixed in 56818fde2fe2cb777fe06cd30f941fa31bb1da0b, release soon.

To try it:

```sh
# npm
npm install keen-ui@JosephusPaye/Keen-UI#next

# yarn
yarn add keen-ui@JosephusPaye/Keen-UI#next
```

See the [changelog](https://github.com/JosephusPaye/Keen-UI/blob/next/CHANGELOG.md) for all changes.

**Comment by JosephusPaye** [comment]
Fix released in [v1.1.0](https://github.com/JosephusPaye/Keen-UI/releases/tag/v1.1.0).
